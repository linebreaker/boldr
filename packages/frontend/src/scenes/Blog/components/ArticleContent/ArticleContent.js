/* @flow */
import React from 'react';
import classnames from 'classnames';
import Divider from '@boldr/ui/Divider';
import Paper from '@boldr/ui/Paper';
import { StyleClasses } from '../../../../theme/styleClasses';
import ArticleDate from '../ArticleDate';

export type Props = {
  content: string,
  createdAt: Date,
  className?: string,
};

const BASE_ELEMENT = StyleClasses.ARTICLE_CONTENT;

const ArticleContent = (props: Props) => {
  const classes = classnames(BASE_ELEMENT, props.className);
  function createMarkup() {
    return {
      __html: props.content,
    };
  }
  return (
    <article>
      <Paper zDepth={2}>
        <ArticleDate created={props.createdAt} />
        <Divider />
        <div className={classes} dangerouslySetInnerHTML={createMarkup()} />
      </Paper>
    </article>
  );
};

export default ArticleContent;
