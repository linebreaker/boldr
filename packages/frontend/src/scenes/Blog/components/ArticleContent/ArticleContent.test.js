import React from 'react';
import { createStore } from 'redux';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ArticleContent from './ArticleContent';

describe('<ArticleContent />', () => {
  it('renders <ArticleContent /> without breaking', () => {
    const wrapper = shallow(<ArticleContent />);
    expect(wrapper.find('article').length).toBe(1);
  });

  it('accepts props and renders them.', () => {
    const wrapper = shallow(
      <ArticleContent feature_image="http://boldr.io/images/logo.png" title="Test Post" />,
    );
    expect(wrapper.instance().props.feature_image).toBe('http://boldr.io/images/logo.png');
    expect(wrapper.instance().props.title).toBe('Test Post');
  });
  it('renders snapshot', () => {
    const wrapper = shallow(<ArticleContent />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
