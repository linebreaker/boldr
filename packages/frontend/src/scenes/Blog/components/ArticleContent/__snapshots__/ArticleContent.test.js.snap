// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleContent /> renders snapshot 1`] = `
<article>
  <Paper
    component="div"
    noPadding={false}
    raiseOnHover={false}
    rounded={false}
    zDepth={2}
  >
    <ArticleDate />
    <Divider />
    <div
      className="boldrui-article__content"
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
    />
  </Paper>
</article>
`;
